### ISSUE TITLE
[Bug] React 18 is not exported from package @storybook/react-dom-shim

### ISSUE THREAD

**Issue by robertwt7** [storybookjs/addon-react-native-web#93]
### Describe the bug

I've been trying to figure out how to setup storybook with react native web but I have no luck.

What's more confusing is that `@storybook/react-native` at the moment is up until version `7.6.x` instead of v8. Does this mean that all the other storybook addons i installed should use v8? If that's the case, will the addons still work on the ondevice sb?

I've tried a bunch of steps here:
1. Pinning sb and all sb addons to v7, i get weird error of not being able to resolve `node-logger`
2. Pinning everything to v8, I also get weird error of:
```
ERROR in ./node_modules/@storybook/react/dist/chunk-SXKPGB5T.mjs 4:0-74
Module not found: Error: Package path ./dist/react-18 is not exported from package /Dev/myapp/node_modules/@storybook/react-dom-shim (see exports field in /Dev/myapp/node_modules/@storybook/react-dom-shim/package.json)
```

After checking the `node_modules/@storybook/react-dom-shim/package.json`, the export is there. So i'm not sure what is wrong with the storybook.

I was also following the migration guide from storybookjs/react-native/MIGRATION.md#from-version-65x-to-76x

This is giving me 2 storybook, folders. I am running the one from `.storybook-web`.
The command that I am running is:
```
"Storybook:web": "storybook dev -p 6006 -c .storybook-web",
```

### Steps to reproduce the behavior

1. Install all the following package from my screenshot
2. Have the same main.ts in the `.storybook-web` follder
3. try running it with 1 simple story 

### Expected behavior

Storybook web is running as expected

### Environment

 - OS: Mac
 - Node.js version: v20.10.0
 - NPM version: v10.2.3
 - Browser (if applicable): Chrome
 - Device (if applicable): macOS

**Comment by robertwt7** [comment]
![image](https://github.com/user-attachments/assets/1a154170-80fa-493f-8f74-e56c552ed818)

![image](https://github.com/user-attachments/assets/2ce5092f-2ab3-408a-88f6-916a71062932)

**Comment by shilman** [comment]
@dannyhw can confirm, but I suspect if you're running both RN and RNW storybooks side by side, they have to be on 7.6 across the board. Storybook RN 8.0 has not been released yet.

**Comment by robertwt7** [comment]
Thanks for your reply. I have tried pinning everything to v7. However some weird error happened (amongst other errors) of `node-logger` not being exported / installed in some sb packages. I've spent too much effort trying to fix it until I upgraded everything to v8. In the end I searched how people managed to use this package across github (including Danny's expo example project) and finally I came across gluestack/gluestack-ui where they have storybook with RN web setup.

After copying similar setup to them, setting everything related to the web to V8 and only having `@storybook/react-native` to v7. Now it works

I just wished this was documented somewhere because the error message wasn't really helping..
